port/unix/omrsignal.c

```diff
@@ -1587,19 +1587,21 @@ initializeSignalTools(OMRPortLibrary *portLibrary)
 
 #if !defined(J9ZOS390)
 #if defined(OSX)
-	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately. */
-	portLibrary->str_printf(portLibrary, semNames[0], 128, "/omr/wakeUpASyncReporter-%d", getpid());
-#define SIGSEM_NAME(_i) semNames[_i]
+	/* OSX only has named semaphores. They are not shared across processes, so unlink immediately.
+	 * The semaphore name length must fit within the SEM_NAME_LEN (31) limit.
+	 */
+	portLibrary->str_printf(portLibrary, semName, sizeof(semName), "/omr-wUASRptr%x-%x", (uint32_t)getpid(), (uint32_t)portLibrary->time_nano_time(portLibrary));
```

Comment:
The tiniest of suggestions: remove 'ptr' from the name, the rest of 'wuasr' is an acronym.

Reply:
Like so?

Reply:
Yes.

Reply:
I don't think there is anything left to address, so waiting for your approval or further review comments.